x/campaign/keeper/msg_server_add_shares.go

```diff
@@ -18,12 +18,19 @@ func (k msgServer) AddShares(goCtx context.Context, msg *types.MsgAddShares) (*t
 		return nil, sdkerrors.Wrapf(types.ErrCampaignNotFound, "%d", msg.CampaignID)
 	}
 
-	// Get the coordinator ID
-	coordinatorID, found := k.profileKeeper.CoordinatorIDFromAddress(ctx, msg.Coordinator)
+	// Get the coordinator ID associated to the sender address
+	coordByAddress, found := k.profileKeeper.GetCoordinatorByAddress(ctx, msg.Coordinator)
 	if !found {
 		return nil, sdkerrors.Wrap(profiletypes.ErrCoordAddressNotFound, msg.Coordinator)
 	}
-	if campaign.CoordinatorID != coordinatorID {
+
+	coord, _ := k.profileKeeper.GetCoordinator(ctx, coordByAddress.CoordinatorID)
```

Comment:
We must handle the case where the coordinator is not found (while the `CoordinatorByAddress` exists) with a critical error

Reply:
Addressed with `GetActiveCoordinatorByAddress()`

Reply:
This PR keeprs `GetCoordinator` and also adds `GetActiveCoordinatorByAddress` which returns `ErrCoordInactive` if the coordinator is found inactive.